esphome/components/logger/logger_zephyr.cpp

```diff
@@ -0,0 +1,83 @@
+#ifdef USE_ZEPHYR
+
+#include "logger.h"
+#include "esphome/core/log.h"
+#include "esphome/core/application.h"
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/uart.h>
+#include <zephyr/usb/usb_device.h>
+
+namespace esphome {
+namespace logger {
+
+static const char *const TAG = "logger";
+
+void Logger::loop() {
+  if (this->uart_ != UART_SELECTION_USB_CDC || nullptr == uart_dev_) {
```

Comment:
Please make sure any instance members are prefixed with `this->`
```suggestion
  if (this->uart_ != UART_SELECTION_USB_CDC || nullptr == this->uart_dev_) {
```